_static/link_gen/link.js

```diff
@@ -100,37 +113,37 @@ var apps = {
     }
 }
 
+function clearLinks(){
+    document.getElementById('default-link').value = "";
+    document.getElementById('binder-link').value = "";
+    document.getElementById('canvas-link').value = "";
+}
+
 function changeTab(div) {
     var hub = document.getElementById("hub");
     var hub_help_text = document.getElementById("hub-help-text");
-    var env_repo = document.getElementById("repo");
-    var env_repo_branch = document.getElementById("branch");
-    var env_repo_help_text = document.getElementById("env-repo-help-text");
-    var content_repo = document.getElementById("content-repo-group");
-    var content_branch = document.getElementById("content-branch-group");
+    var env_repo_group = document.getElementById("env-repo-group");
+    var env_repo = document.getElementById("env-repo");
```

Comment:
I understand you have renamed the element id's to match the variable names - okay, it was a bit out of scope for this PRs main purpose and makes the review and tracking changes a bit trickier, but sure I appreciate the consistency.

Removed: `content_repo` `content_branch` env_repo_help_text` and `env_repo_branch`, 
Added: `env_repo_group`

Ah... You have refactored so you hide an entire `<div>` of things in one go? Okay makes sense.

Reply:
ok I hear you here. It got a bit tricky to follow the patterns in these pages which inspired me to make things clearer at least to myself! I can see how that makes the PR review more difficult. I will make sure to separate out refactoring commits from commits directly related to the PR.

Reply:
The main problem I ran into with the original code was that we were changed the meaning of repoX(e.g. repoName, repoURL, etc)  depending on the context which caused me great headaches!  In other words, if you were NOT on the binder tab then repoX meant the content repo where the notebooks are stored. But if you were on the binder tab, then repoX meant the environment git repo you are using and contentRepoX meant the repo with the notebooks. It killed me when we went to add the non-git sources to the binder tab. 

You will see logic in the next commit that removes the need a conditional we used to generate the urlPath because of this situation(link.js - line 245)